vowpalwabbit/cb_explore_adf.cc

```diff
@@ -575,10 +514,14 @@ base_learner* cb_explore_adf_setup(arguments& arg)
       arg.trace_message << "warning: cb_type must be in {'ips','dr'}; resetting to ips." << std::endl;
   }
 
-  learner<cb_explore_adf>& l = init_learner(data, base, CB_EXPLORE_ADF::predict_or_learn<true>, CB_EXPLORE_ADF::predict_or_learn<false>, problem_multiplier, prediction_type::action_probs);
+  learner<cb_explore_adf,multi_ex>& l = init_learner(data, base,
+    CB_EXPLORE_ADF::do_actual_learning<true>,
+    CB_EXPLORE_ADF::do_actual_learning<false>,
+    problem_multiplier, 
+    prediction_type::action_probs);
 
   l.set_finish_example(CB_EXPLORE_ADF::finish_multiline_example);
   l.set_finish(CB_EXPLORE_ADF::finish);
-  l.set_end_examples(CB_EXPLORE_ADF::end_examples);
+  l.set_test_example(CB::example_is_test);
```

Comment:
Why is this needed?

Reply:
Todo: sort out so is_test is a part of the label type.

Reply:
Defered to John's upcoming change